### PR TITLE
Clipping, slicing and normalizing

### PR DESCRIPTION
**Pull request by smattis**
Adds clipping of sample sets and discretizations, slicing of discretizations, and normalizing and unormalizing sample sets.

### CODE REVIEW

**Comment by smattis** [comment]
@lcgraham Should be all ready for review.

File: test/test_sample.py [hunk 1]
```diff
@@ -21,6 +21,24 @@ def setUp(self):
...
+    def test_normalize(self):
+        """
+        Test normalize and undo normalize domain.
+        """
+        domain = 5.0*self.domain -1.0 
+        self.sam_set.set_domain(domain)
+        self.sam_set.normalize_domain()
+        nptest.assert_array_equal(self.sam_set._domain, self.domain)
+        self.sam_set.undo_normalize_domain()
+        nptest.assert_array_equal(self.sam_set._domain, domain)
```
Comment (lcgraham) [inline test/test_sample.py]: Need to check attributes that get altered besides just the domain.

File: test/test_sample.py [hunk 2]
```diff
@@ -21,6 +21,24 @@ def setUp(self):
...
+    def test_clip(self):
+        """
+        Test clipping of sample set.
+        """
+        cnum = int(0.5*self.num)
+        sam_set_clipped = self.sam_set.clip(cnum)
+        nptest.assert_array_equal(self.sam_set._values[0:cnum,:],
```
Comment (lcgraham) [inline test/test_sample.py]: Need to check reference value and Jacobians.

File: test/test_sample.py [hunk 3]
```diff
@@ -410,7 +428,26 @@ def Test_check_nums(self):
...
+        """
+        cnum = int(0.5*self.num)
+        disc_clipped = self.disc.clip(cnum)
+        nptest.assert_array_equal(self.disc._input_sample_set._values[0:cnum,:],
+                                  disc_clipped._input_sample_set._values)
+        nptest.assert_array_equal(self.disc._output_sample_set._values[0:cnum,:],
+                                  disc_clipped._output_sample_set._values)
+    def Test_slicing(self):
+        """
+        Test `bet.sample.discretization.choose_inputs_outputs`
```
Comment (lcgraham) [inline test/test_sample.py]: Need to check reference values and Jacobians since they also get sliced.

**Comment by lcgraham** [comment]
Looks good, minor changes to tests.

**Comment by smattis** [comment]
@lcgraham added stuff to the tests

**Pull request merged**